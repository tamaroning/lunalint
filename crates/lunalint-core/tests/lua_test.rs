mod helper;

use std::path::PathBuf;
use strip_ansi_escapes::strip_str;

use insta::Settings;

macro_rules! ident_to_str {
    ($ident:ident) => {
        stringify!($ident).replace("_", "-")
    };
}

macro_rules! lua_test {
    ($name:ident) => {
        #[test]
        fn $name() {
            let name = ident_to_str!($name);
            let path = PathBuf::from(format!("tests/lua/{}.lua", name));
            let out = strip_str(helper::run_linter(&path));
            let mut settings = Settings::new();
            settings.set_prepend_module_to_snapshot(false);
            settings.set_omit_expression(true);
            settings.bind(|| insta::assert_snapshot!(out));
        }
    };
}

lua_test!(undefined_global);
lua_test!(parse_error);
