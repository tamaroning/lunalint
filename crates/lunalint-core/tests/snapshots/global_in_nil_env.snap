---
source: crates/lunalint-core/tests/lua_test.rs
---
Error: Invalid global (`_ENV` is `nil`) (global-in-nil-env)
   ╭─[tests/lua/global-in-nil-env.lua:7:1]
   │
 7 │ _ENV = nil
   │ ──┬─  
   │   ╰─── Assignment occurs here
   │ 
   │ Help: for further information visit https://luals.github.io/wiki/diagnostics/#global-in-nil-env
───╯
